packages/ipfs-core/src/components/index.js

```diff
@@ -241,7 +242,7 @@ const addEmptyDir = async (ipfs) => {
 
   const cid = await ipfs.block.put(buf, {
     mhtype: 'sha2-256',
-    format: 'dag-pb'
+    storeCodec: 'dag-pb'
```

Comment:
Is this done on purpose?  There's been no change to `ipfs.block.put` in `ipfs-core` to support a `storeCodec` options.

Reply:
An aside, but it's Very Tedious that the type checker doesn't catch this.

Reply:
Aha, we can make it catch it.  In `packages/ipfs-core/src/componenets/block/put.js`, change:

```js
  /**
   * @type {import('ipfs-core-types/src/block').API["put"]}
   */
  async function put (block, options = {}) {
```

to:

```js
  /**
   * @type {import('ipfs-core-types/src/block').API<{}>["put"]}
   */
  async function put (block, options = {}) {
```

then we get:

```
src/components/index.js:245:5 - error TS2345: Argument of type '{ mhtype: string; storeFormat: string; }' is not assignable to parameter of type 'PutOptions'.
  Object literal may only specify known properties, and 'storeFormat' does not exist in type 'PutOptions'.

245     storeFormat: 'dag-pb'
        ~~~~~~~~~~~~~~~~~~~~~
```

JSDoc doesn't pick up the default option extension in `/packages/ipfs-core-types/src/componenets/block/index.ts`.  That's annoying

Reply:
https://github.com/ipfs/js-ipfs/pull/3965

Reply:
Merged into this branch, all good!